packages/react-widgets/src/widgets/DrawingToolWidget.js

```diff
@@ -0,0 +1,171 @@
+  </SvgIcon>
+);
+
+const RectangleIcon = (
+  <SvgIcon>
+    <path d='M4 18a2 2 0 1 1 0 4 2 2 0 0 1 0-4zm16 0a2 2 0 1 1 0 4 2 2 0 0 1 0-4zm-2.829 1a2.994 2.994 0 0 0-.17.974l-.001.052.007.183c.02.275.075.54.164.79H6.829a2.995 2.995 0 0 0 0-2h10.342zM2.998 6.828a2.995 2.995 0 0 0 2.002 0v10.342a2.995 2.995 0 0 0-2.002 0V6.83zm16.001 0a2.995 2.995 0 0 0 2 0v10.342a2.995 2.995 0 0 0-2 0V6.829zM20 2a2 2 0 1 1 0 4 2 2 0 0 1 0-4zM4 2a2 2 0 1 1 0 4 2 2 0 0 1 0-4zm13.171.998a2.994 2.994 0 0 0-.17.976L17 4.026l.007.183c.02.275.075.54.164.79H6.829a2.995 2.995 0 0 0 0-2H17.17z' />
+  </SvgIcon>
+);
+
+const CircleIcon = (
+  <SvgIcon>
+    <path d='M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 1 0 0 16 8 8 0 0 0 0-16zm0 6a2 2 0 1 1 0 4 2 2 0 0 1 0-4z' />
+  </SvgIcon>
+);
+
+const LassoIcon = (
+  <SvgIcon>
+    <path d='M12 3c4.935 0 9 3.736 9 9l-.002.343-.012.669c-.012.437-.033.86-.062 1.265l-.05.597C20.498 18.767 19.267 21 17 21c-1.192 0-1.971-.341-2.987-1.122l-.473-.375c-.401-.319-.64-.473-.888-.566a4.894 4.894 0 0 0-.415-.13l-.34-.085-.398-.086-.456-.086-.66-.111-1.708-.273a9.103 9.103 0 0 1-.952-.206C5.46 17.301 3 14.954 3 12.015c0-1.508.485-2.995 1.436-4.458.355.585.906 1.04 1.562 1.272C5.328 9.916 5 10.977 5 12.015c0 1.889 1.78 3.588 3.282 4.025l.085.023.345.076.517.092 1.619.257.583.1.518.099.237.05.433.102c.272.07.512.143.73.224.434.161.783.373 1.235.718l.457.362c.806.646 1.24.857 1.959.857.893 0 1.63-1.518 1.895-4.45l.045-.585c.013-.2.024-.407.033-.62l.02-.655c.005-.224.007-.454.007-.69 0-4.12-3.134-7-7-7a1 1 0 0 1 0-2zM7 4a2 2 0 1 1 0 4 2 2 0 0 1 0-4z' />
+  </SvgIcon>
+);
+
+const DRAW_MODES_MAP = {
+  [DRAW_MODES_KEYS.POLYGON]: {
+    label: 'Polygon',
+    icon: PolygonIcon
+  },
+  [DRAW_MODES_KEYS.RECTANGLE]: {
+    label: 'Rectangle',
+    icon: RectangleIcon
+  },
+  [DRAW_MODES_KEYS.CIRCLE]: {
+    label: 'Circle',
+    icon: CircleIcon
+  },
+  [DRAW_MODES_KEYS.LASSO_TOOL]: {
+    label: 'Lasso tool',
+    icon: LassoIcon
+  }
+};
+
+const EDIT_MODES_MAP = {
+  [EDIT_MODES_KEYS.EDIT]: {
+    label: 'Edit geometry',
+    icon: CursorIcon
+  }
+};
+
+function DrawingToolWidget({
+  className,
+  drawModes: drawModesKeys,
+  editModes: editModesKeys,
+  defaultActivated,
+  defaultSelectedMode,
+  tooltipPlacement
+}) {
+  const dispatch = useDispatch();
+  const [activated, setActivated] = useState(defaultActivated);
```

Comment:
1. selectedMode is duplicated in redux and here. Not easy to understand why
2. we need to do something odd to skip react-hooks/exhaustive-deps rule
3. activated should be exposed outside since it's required for cursor.

Let's add a new state in redux carto/setDrawingToolEnabled to control the enablement of the drawing tool in a standard way for all the components. So, the layer, the widget or the map (cursor) will behave by checking this unique state drawingToolEnabled

Reply:
Done.